src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -203,12 +279,17 @@ public override bool TryGetPropertyValue(string name, out object value)
                 object deltaNestedResource = _deltaNestedResources[name];
 
                 Contract.Assert(deltaNestedResource != null, "deltaNestedResource != null");
+
+                //If DeltaSet collection, we are handling delta collections so the value will be that itself and no need to get instance value
+                if (deltaNestedResource is IDeltaSet)
+                {
+                    value = deltaNestedResource;
+                    return true;
+                }
+
                 Contract.Assert(IsDeltaOfT(deltaNestedResource.GetType()));
 
                 // Get the Delta<{NestedResourceType}>._instance using Reflection.
-                // This semantic is not correct but is preserved for backwards
```

Comment:
Does this warning still apply? Or is this feature planned for 8.x?

Reply:
This feature should be supported in 7.x.

Reply:
I believe the warning still applies.